classes/UpgradeTools/CoreUpgrader/CoreUpgrader.php

```diff
@@ -689,6 +701,75 @@ protected function updateTheme()
         }
     }
 
+    protected function updateRTLFiles()
+    {
+        if (!$this->container->getUpgradeConfiguration()->shouldUpdateRTLFiles()) {
+            return;
+        }
+
+        // BO theme
+        if (class_exists(RtlStylesheetProcessor::class)) {
+            $this->logger->info($this->container->getTranslator()->trans('Upgrade the RTL files of back-office themes.', [], 'Modules.Autoupgrade.Admin'));
+
+            $this->removeExistingRTLFiles([
+                ['directory' => $this->container->getProperty(UpgradeContainer::PS_ADMIN_PATH) . DIRECTORY_SEPARATOR . 'themes'],
+            ]);
+
+            (new RtlStylesheetProcessor(
+                $this->container->getProperty(UpgradeContainer::PS_ADMIN_PATH),
+                $this->container->getProperty(UpgradeContainer::PS_ROOT_PATH) . DIRECTORY_SEPARATOR . 'themes',
+                []
+            ))
+                ->setProcessBOTheme(true)
+                ->setProcessDefaultModules(true)
+                ->process();
+        }
+
+        // FO themes
+        if (!class_exists(AdaptThemeToRTLLanguagesCommand::class)) {
+            return;
+        }
+
+        $this->logger->info($this->container->getTranslator()->trans('Upgrade the RTL files of front-office themes.', [], 'Modules.Autoupgrade.Admin'));
+        $themeAdapter = new ThemeAdapter($this->db);
+
+        $themes = $themeAdapter->getListFromDisk();
+        $this->removeExistingRTLFiles($themes);
+
+        foreach ($themes as $theme) {
+            $adaptThemeToTRLLanguages = new AdaptThemeToRTLLanguagesCommand(
+                new ThemeName($theme['name'])
+            );
+
+            /** @var CommandBusInterface $commandBus */
+            $commandBus = $this->container->getModuleAdapter()->getCommandBus();
+
+            try {
+                $commandBus->handle($adaptThemeToTRLLanguages);
+            } catch (CoreException $e) {
+                $this->logger->error('
+                    [ERROR] PHP Impossible to generate RTL files for theme' . $theme['name'] . "\n" .
+                    $e->getMessage()
+                );
```

Comment:
We are already in an error log, why add "[ERROR]"?

Reply:
I think this is to make sure the prepended `[ERROR]` is seen by the user. The logger should be able to do it itself but it is not the case, and this code was only moved from CoreUpgrader17.php.